pyqtgraph/functions.py

```diff
@@ -1068,6 +1069,69 @@ def rescaleData(data, scale, offset, dtype=None, clip=None):
     return data
 
 
+def rescaleData_blocked(data_in, scale, offset, dtype=None, clip=None):
```

Comment:
Other than the need for CuPy compatibility, this new version should always be used, yes? This code is meant to completely replace the old version, for all behaviors. Let's make that more explicit, and name things accordingly. We should bring over the docstring at the same time, and we should give some guidance as to which version of `rescaleData` a user should use.

Reply:
Actually, np.nditer would be hardly worth implementing for Cupy. It would be an anti-pattern for CUDA to process block by block.

During the initial implementation, it was indeed meant to be a replacement. However, I can guess from the optional arguments that many functions in pyqtgraph must have grown from their original implementation to add additional functionality. e.g. for ```rescaleData()```, ```clip``` argument must have been added for the levels + lut combine usage in ```ImageItem.py```. Having to add support for all these optional functionalities makes it more difficult to optimize for specific use-cases that do have bottlenecks.